RecoTauTag/HLTProducers/src/L2TauTagNNProducer.cc

```diff
@@ -0,0 +1,887 @@
+            return GetCellImpl(cellGridMatrix, tau_idx, phi_idx, eta_idx, input);
+          };
+          getCell(NNInputs::EcalEnergySum)+=ebCalEn;
+          getCell(NNInputs::EcalSize)+=1.;
+          getCell(NNInputs::EcalEnergyStdDev)+=ebCalEn * ebCalEn;
+          getCell(NNInputs::EcalDeltaEta)+=deta*ebCalEn;
+          getCell(NNInputs::EcalDeltaPhi)+=dphi*ebCalEn;
+          if(ebCalChi2>=0){
+            getCell(NNInputs::EcalChi2)+=ebCalChi2*ebCalEn;
+            getCell(NNInputs::EcalEnergySumForPositiveChi2)+=ebCalEn;
+            getCell(NNInputs::EcalSizeForPositiveChi2)+=1.;
+          }
+        }
+      }
+
+
+      // caorechit_HBHE
+      for (const auto& caloRecHit_hbhe : *caloRecHits.hbhe){
+        if(caloRecHit_hbhe.energy()<=0) continue;
+        const auto& position = caloRecHits.Geometry->getGeometry(caloRecHit_hbhe.id())->getPosition();
+        const float hbheCalEn = caloRecHit_hbhe.energy();
+        const float hbheCalChi2 = caloRecHit_hbhe.chi2();
+        if(reco::deltaR2(position,allTaus[tau_idx]->polarP4())<dR2_max){
+          std::tie(deta, dphi, eta_idx, phi_idx) = getEtaPhiIndices(position, allTaus[tau_idx]->polarP4());
+          auto getCell = [&](NNInputs input) -> float& {
+            return GetCellImpl(cellGridMatrix, tau_idx, phi_idx, eta_idx, input);
+          };
+          getCell(NNInputs::HcalEnergySum)+=hbheCalEn;
+          getCell(NNInputs::HcalEnergyStdDev)+=hbheCalEn * hbheCalEn;
+          getCell(NNInputs::HcalSize)+=1.;
+          getCell(NNInputs::HcalDeltaEta)+=deta*hbheCalEn;
+          getCell(NNInputs::HcalDeltaPhi)+=dphi*hbheCalEn;
+          if(hbheCalChi2>=0){
+            getCell(NNInputs::HcalChi2)+=hbheCalChi2*hbheCalEn;
+            getCell(NNInputs::HcalEnergySumForPositiveChi2)+=hbheCalEn;
+            getCell(NNInputs::HcalSizeForPositiveChi2)+=1.;
+          }
+        }
+      }
+
+      // caorechit_HO
+      for (const auto& caloRecHit_ho : *caloRecHits.ho){
+        if(caloRecHit_ho.energy()<=0) continue;
+        const auto& position = caloRecHits.Geometry->getGeometry(caloRecHit_ho.id())->getPosition();
+        const float hoCalEn = caloRecHit_ho.energy();
+        if(reco::deltaR2(position,allTaus[tau_idx]->polarP4())<dR2_max){
+          std::tie(deta, dphi, eta_idx, phi_idx) = getEtaPhiIndices(position, allTaus[tau_idx]->polarP4());
+          auto getCell = [&](NNInputs input) -> float& {
+            return GetCellImpl(cellGridMatrix, tau_idx, phi_idx, eta_idx, input);
+          };
+          getCell(NNInputs::HcalEnergySum)+=hoCalEn;
+          getCell(NNInputs::HcalEnergyStdDev)+=hoCalEn * hoCalEn;
+          getCell(NNInputs::HcalSize)+=1.;
+          getCell(NNInputs::HcalDeltaEta)+=deta*hoCalEn;
+          getCell(NNInputs::HcalDeltaPhi)+=dphi*hoCalEn;
+        }
+      }
+
+      // normalize to sum and define stdDev
+      for (int eta_idx = 0; eta_idx<nCellEta ; eta_idx++){
+        for (int phi_idx = 0; phi_idx<nCellPhi ; phi_idx++){
```

Comment:
use `eta_idx` and `phi_idx` defined above. I.e. `for(eta_idx = 0; eta_idx < nCellEta; eta_idx++)` etc.